### PR TITLE
Share vaccine homepage button & content

### PR DESCRIPTION
**Pull request by bionara**
# Description

Homepage banner to share an image for vaccine promotion (note: shared image[png] is slightly diff to the button[svg])

Exists on UK and US - same content

Tapping the image opens the native share sheet (at the bottom) and not a new popup screen with image/share button.

Need to test on device to check sharing output.

`onShareImage` here is quite nifty and could be extracted into a reusable component: it copies the image to the user's local lib, then pushes itself as a shared item from there.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

image
image

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

**Comment by selvach** [comment]
Code looks great! Could you confirm that an analytics even is fired - it looks like it's missing? You may want to check the other share buttons.

File: src/features/dashboard/DashboardLogVaccine.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import * as FileSystem from 'expo-file-system';
+import React from 'react';
+import { Image } from 'react-native';
+import Svg, { Path, Rect, G, Defs, ClipPath } from 'react-native-svg';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { shareVaccine } from '@assets';
+import i18n from '@covid/locale/i18n';
+
+interface Props {}
```
Comment (petermason) [inline src/features/dashboard/DashboardLogVaccine.tsx:11]: not required

File: src/features/dashboard/DashboardLogVaccine.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+export const DashboardLogVaccine: React.FC<Props> = ({}) => {
```
Comment (petermason) [inline src/features/dashboard/DashboardLogVaccine.tsx:13]: * can be written as function() which is cleaner. No need for React.FC<Props>
* there are no props

File: src/features/dashboard/DashboardLogVaccine.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  };
+
+  return (
+    <TouchableOpacity style={{ marginLeft: -14, marginTop: -30 }} onPress={onShareImage}>
+      {svgButton}
+    </TouchableOpacity>
+  );
+};
+
+const svgButton = (
```
Comment (petermason) [inline src/features/dashboard/DashboardLogVaccine.tsx:34]: is this a one off or should it be a shared asset?

File: src/features/dashboard/DashboardLogVaccine.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import i18n from '@covid/locale/i18n';
+
+interface Props {}
+
+export const DashboardLogVaccine: React.FC<Props> = ({}) => {
+  const onShareImage = async () => {
+    try {
+      const uri = Image.resolveAssetSource(shareVaccine).uri;
+      const downloadPath = FileSystem.cacheDirectory + 'ShareVaccine.png';
+      const { uri: localUrl } = await FileSystem.downloadAsync(uri, downloadPath);
```
Comment (petermason) [inline src/features/dashboard/DashboardLogVaccine.tsx:18]: nice

**Comment by bionara** [comment]
> Code looks great! Could you confirm that an analytics even is fired - it looks like it's missing? You may want to check the other share buttons.

whoops, that wasn't in the ticket sorry. added GET_VACCINE_DISPLAYED and GET_VACCINE_SHARED for button view and share

File: src/features/dashboard/DashboardLogVaccine.tsx (hunk 2 above)
Comment (bionara) [reply src/features/dashboard/DashboardLogVaccine.tsx:13]: Cheers! ironically this now needs a prop - id - for the analytics :) will bare in mind the empty ver for next time tho thanks

File: src/features/dashboard/DashboardLogVaccine.tsx (hunk 3 above)
Comment (bionara) [reply src/features/dashboard/DashboardLogVaccine.tsx:34]: one off for this work - it i the entire button that sits on the dashboard

**Review by petermason — DISMISSED** [review]

File: src/features/dashboard/DashboardLogVaccine.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import { Image } from 'react-native';
+import Svg, { Path, Rect, G, Defs, ClipPath } from 'react-native-svg';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import Analytics, { events } from '@covid/core/Analytics';
+import { shareVaccine } from '@assets';
+import i18n from '@covid/locale/i18n';
+
+interface Props {
+  id: string;
```
Comment (selvach) [inline src/features/dashboard/DashboardLogVaccine.tsx:13]: could you rename this screenName please

**Review by selvach — CHANGES_REQUESTED** [review]

File: src/core/Analytics.ts [hunk 6]
```diff
@@ -34,6 +34,8 @@ const DashboardEvents = {
   TRENDLINE_EXPLORE_SHARE_CLICKED: 'TRENDLINE_EXPLORE_SHARE_CLICKED',
   DIET_STUDY_PLAYBACK_CLICKED: 'DIET_STUDY_PLAYBACK_CLICKED',
   DIET_STUDY_PLAYBACK_DISPLAYED: 'DIET_STUDY_PLAYBACK_DISPLAYED',
+  GET_VACCINE_SHARED: 'GET_VACCINE_SHARED',
```
Comment (selvach) [inline src/core/Analytics.ts:37]: Rename to `LOG_YOUR_VACCINE` please!

File: src/core/Analytics.ts [hunk 7]
```diff
@@ -34,6 +34,8 @@ const DashboardEvents = {
...
+  GET_VACCINE_DISPLAYED: 'GET_VACCINE_DISPLAYED',
```
Comment (selvach) [inline src/core/Analytics.ts:38]: I don't think we need the LOG_YOUR_VACCINE_DISPLAYED event - as it' shown to everyone!

File: src/features/dashboard/DashboardLogVaccine.tsx (hunk 5 above)
Comment (bionara) [reply src/features/dashboard/DashboardLogVaccine.tsx:13]: done

**Review by selvach — APPROVED** [review]

**Comment by selvach** [comment]
FYI - I just renamed the screenName to remove the "Screen" at the end. In the analytics this isn't there - we use the name in `ScreenParamList`

**Pull request merged**